File: app/src/main/java/mohsin/reza/propertyapp/db/PropertyDao.java
package mohsin.reza.propertyapp.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import mohsin.reza.propertyapp.vo.Property;

@Dao
public abstract class PropertyDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public abstract void insert(List<Property> propertyList);


    @Query("Select * from Property")
    public abstract LiveData<List<Property>> loadPropertyList();
}

File: app/src/main/java/mohsin/reza/propertyapp/db/PropertyDao.kt
package mohsin.reza.propertyapp.db

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

import mohsin.reza.propertyapp.vo.Property

@Dao
abstract class PropertyDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    abstract fun insert(propertyList: List<Property>)


    @Query("Select * from Property")
    abstract fun loadPropertyList(): LiveData<List<Property>>
}
